### ISSUE TITLE
Visualize addressability of ArrayPort connections

### ISSUE THREAD

**Issue by bergie** [noflo/noflo-ui#14]
Currently we show an ArrayPort in NoFlo UI simply as a port that can have multiple connections.

However, this doesn't show the fact that ArrayPort connections are individually addressable (see noflo/noflo#90). So instead of showing a single `IN` port, we should show:
- `IN[0]`
- `IN[1]`

etc.

The component instance should always show one free port, so when you attach a connection, a new port should appear below it.

**Comment by bergie** [comment]
Visual example where this would be nice. With `xpress/Router` the incoming patterns, and outgoing requests are matched by arrayport index.

<img width="511" alt="screen shot 2015-07-21 at 19 20 26" src="https://cloud.githubusercontent.com/assets/3346/8807604/f292d59c-2fdd-11e5-8807-6e7789927847.png">

**Comment by automata** [comment]
Thinking about a vertical x horizontal way to visualize that:

![img_20150721_151334](https://cloud.githubusercontent.com/assets/49062/8809174/4d57f794-2fbd-11e5-8e44-a225aaf8f9b5.jpg)

**Comment by bergie** [comment]
@automata that might be interesting for space-saving!

Original plan looked like this:

![ArrayPorts](https://camo.githubusercontent.com/660dfa0ffefd855ef44bed646bc26ffd5df79c13/68747470733a2f2f662e636c6f75642e6769746875622e636f6d2f6173736574732f333334362f3734333130312f33326266666361342d653365302d313165322d393537612d6465316630373063633533622e706e67)

**Comment by jonnor** [comment]
I at least think it should respect level-of-detail as set by the zoom level. Zoomed out far it should be one port, up close it should explode out.

**Comment by alfa256** [comment]
I agree with one distinct inport/outport per connection like I have in my visualizations:
http://i.imgur.com/4LQdYfl.png
